docs/production/authentication-methods.md

```diff
@@ -518,6 +518,65 @@ to the root and `engineering` subdomains:
 </saml2:Attribute>
 ```
 
+### Using Keycloak as a SAML IdP
+
+1. Make sure you reviewed [this article][saml-help-center], which
+   details how to configure Keycloak properly to use SAML with Zulip.
+2. Verify that `SOCIAL_AUTH_SAML_ENABLED_IDPS['entity_id']` and
+   `SOCIAL_AUTH_SAML_ENABLED_IDPS['url']` are correct in your Zulip
+   configuration. Specifically, if `entity_id` is
+   `https://keycloak.example.com/auth/realms/master`, then `url`
+   should be
+   `https://keycloak.example.com/auth/realms/master/protocol/saml`
+3. Your Keycloak public certificate must be saved on the Zulip server
+   as `{idp_name}.crt` in `/etc/zulip/`. You can obtain the
+   certificate from the Keycloak UI in the `Keys` tab.  Click on the
+   button `Certificate` and copy the content.
+
+   (Alternatively, open the URL in your browser
+   `https://keycloak.example.com/auth/realms/master/protocol/saml/descriptor`.
+   Replace the domain (`keycloak.example.com`) as well as the realm
+   name (`master`) in the url. The certificate is the content inside
+   `<ds:X509Certificate>[...]</ds:X509Certificate>`).
+
+   Save the certificate in an `{idp_name}.crt` file constructed as follows:
```

Comment:
```suggestion
   Save the certificate in the `{idp_name}.crt` file constructed as follows:
```
is the correct grammar here i think

Reply:
Maybe "in a new" -- we should make clear they're creating a new file here.